### PR TITLE
XS-2873: Update the documentation for DQC

### PR DESCRIPTION
**Pull request by chrislococo-wf**
### Changes:

- Added a `.github` folder with a PR template and placeholder for issue and contributing templates.
- Updated the README.md to document the software development lifecycle.
- Removed PR template from the README.md, as it will now auto-populate once this ticket merges.
- Added the QA Template in order to standardize the QA process for the library.

### Pull request process

- [x] Full description of changes provided above.
- [x] Reviewed by another person. (+1)
- [x] If a code change, reviewed by a second person. (+1)
- [ ] QA'd to the specifications listed [here](https://github.com/Workiva/dqc_us_rules#quality-assurance-qa-of-a-pull-request). Documentation only changes do not require the formal code quality assurance process. (+10)

##### Once checklist is complete, @hefischer  @andrewperkins-wf please review for merge.

### CODE REVIEW

**Comment by chrislococo-wf** [comment]
Easy to view links:
PR Template: https://github.com/chrislococo-wf/dqc_us_rules/blob/XS-2873/.github/PULL_REQUEST_TEMPLATE.md
README.md updates: https://github.com/chrislococo-wf/dqc_us_rules/tree/XS-2873#proposed-changes

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+### Changes:
+
+- Change_one with reasoning
+- Change_two with reasoning ...
+
+### Pull request process
+
+- [ ] Full description of changes provided above.
+- [ ] Reviewed by another person. (+1)
+- [ ] If a code change, reviewed by a second person. (+1)
```
Comment (sagesmith-wf) [inline .github/PULL_REQUEST_TEMPLATE.md]: Maybe:
```- [ ] If coded changed, reviewed by a second person. (+1)```

File: README.md [hunk 2]
```diff
@@ -65,24 +65,51 @@ The rule definition index is [here](docs/README.md).
...
 
-### Requirements for a Pull Request(PR):
+### Development of Rules that are "Ready for Coding":
+
+When new rules that have been approved for coding are released by the DQC, the rules will be developed on a branch named `next_q#_YY` where the `#` is the quarter, and the `YY` is replaced by the current year. All new coding for the proposed rules will target this branch on the root DataQualityCommittee fork. Periodically, this branch will be tagged and released on the global pypi index as a release candidate (RC). Once final approval for the rules is complete, the RC versions of the library will be removed from the index, the next branch merged into master, and a new major version of the library will be released on the [global pypi index](https://pypi.python.org/simple/dqc-us-rules/).
```
Comment (sagesmith-wf) [inline README.md]: ,the next branch *will* be merged into master,

File: README.md [hunk 3]
```diff
@@ -65,24 +65,51 @@ The rule definition index is [here](docs/README.md).
...
-    - Change_one with reasoning
-    - Change_two with reasoning
-
-    Please review: @hefischer  @andrewperkins-wf
-    ```
+  - The pull request must go through the review process described below.
+
+### Pull Request Review Process:
+
+  - Each pull request must have at least one `+1` from another community member.
```
Comment (sagesmith-wf) [inline README.md]: `+1` comment

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (chrislococo-wf) [inline .github/PULL_REQUEST_TEMPLATE.md]: Do you mean "if code changed"?

File: README.md [hunk 4]
```diff
@@ -65,24 +65,51 @@ The rule definition index is [here](docs/README.md).
...
+  - For code changes, you must have a second `+1` from yet another community member.
```
Comment (sagesmith-wf) [inline README.md]: `+1` comment from a second community member.

File: README.md (hunk 2 above)
Comment (chrislococo-wf) [inline README.md]: Parallelism, nice catch

File: README.md [hunk 5]
```diff
@@ -65,24 +65,51 @@ The rule definition index is [here](docs/README.md).
...
+  - The request will need to go through the Quality Assurance process defined below and receive a `+10`. This can be from any other community member, including one of the reviewers.
```
Comment (sagesmith-wf) [inline README.md]: ``` `+10` comment if the request is correctly executed```

**Comment by sagesmith-wf** [comment]
You could add a section about adding a pre-push git hook to run flake8 and the unit tests:

i.e. .git/hooks/pre-push

!#/bin/bash
flake8 .
nosetests tests

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (sagesmith-wf) [inline .github/PULL_REQUEST_TEMPLATE.md]: Yea something like that

**Comment by chrislococo-wf** [comment]
Honestly, I am not sure what the technical level would be of people developing on this, and not everyone uses git hooks, so I would avoid explicitly demanding them. The readme is for a weird mix of developers and not developers, so I am trying to avoid being too low level. What do you think @andrewperkins-wf ?

**Comment by andrewperkins-wf** [comment]
Hmm, I think that if we want to add something like pre-commit hook information, it would probably be a development help section or document.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+- Change_one with reasoning
+- Change_two with reasoning ...
+
+### Pull request process
+
+- [ ] Full description of changes provided above.
+- [ ] Reviewed by another person. (+1)
+- [ ] If this is a code change, reviewed by a second person. (+1)
+- [ ] QA'd to the specifications listed [here](https://github.com/Workiva/dqc_us_rules#quality-assurance-qa-of-a-pull-request). Documentation only changes do not require the formal code quality assurance process. (+10)
```
Comment (andrewperkins-wf) [inline .github/PULL_REQUEST_TEMPLATE.md]: This should not be pointing to a workiva repository

**Comment by andrewperkins-wf** [comment]
@chrislococo-wf  One other thing we need: a discussion of versioning.  Specifically, that rules will have the version number associated with them that was the last time they were substantially changed, e.g. their behavior was modified, either through configuration or code.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 6 above)
Comment (chrislococo-wf) [inline .github/PULL_REQUEST_TEMPLATE.md]: Doh, nice catch, thank you

**Comment by sagesmith-wf** [comment]
+1

**Comment by mindeerickson-wf** [comment]
+1

**Pull request merged**